core/domain/models/command.go

```diff
@@ -70,6 +70,27 @@ func (c Command) String() string {
 	return string(out)
 }
 
+func(c *Command) UnmarshalJSON(b []byte) error {
```

Comment:
The approach below was taken from the following article and is utilized in order to prevent an endless loop when calling json.Unmarshal().

http://choly.ca/post/go-json-marshalling/